src/Testcontainers/Builders/TlsEndpointAuthenticationProvider.cs

```diff
@@ -0,0 +1,92 @@
+namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.IO;
+  using System.Linq;
+  using System.Net.Security;
+  using System.Security.Cryptography.X509Certificates;
+  using DotNet.Testcontainers.Configurations;
+
+  /// <inheritdoc cref="IDockerRegistryAuthenticationProvider" />
+  internal class TlsEndpointAuthenticationProvider : DockerEndpointAuthenticationProvider
+  {
+    private const string DefaultUserDockerFolderName = ".docker";
+    private const string DefaultCaCertFileName = "ca.pem";
```

Comment:
Is `ca.pem` correct here? If I look into the Docker [docs](https://docs.docker.com/engine/security/protect-access/#other-modes) the client mode does not mention `tlscacert` for the TLS configuration.

Reply:
You are right - I run a lot of test with Docker CLI and CA is required for validation of the server certificate. I used it for current implementation.
1. try to validate cert against system CAs store
2. used ca.pem folder as CLI does (even docs said something else)

Reply:
Based on [documentation](https://docs.docker.com/engine/security/protect-access/#other-modes) where client combinations are mentioned:

> Client modes
> tls: Authenticate server based on public/default CA pool
> tlsverify, tlscacert: Authenticate server based on given CA
> tls, tlscert, tlskey: Authenticate with client certificate, do not authenticate server based on given CA
> tlsverify, tlscacert, tlscert, tlskey: Authenticate with client certificate and authenticate server based on given CA

Then I understand that documentation this way:
* tls - try look for certificate in system trusted CA certificates storage ("based on public") and then try to use `ca.pem` file ("default CA") but to find one valid CA is enough to pass
* tlsverify, tlscacert - server is still set with `tls=true` only and in this case client verifies server certifcate against one specific CA stored in specified file or in default file which is `ca.pem`
* tls, tlscert, tlskey - tls is validated as is described in first bullet point, and client send cert for authetication because server requires it
* tlsverify, tlscacert, tlscert, tlskey - tlsverify is validated as is described in second bullet point, and client send cert for authetication because server requires it

So if system trusted root CAs storage does not contain any valid ceritificate and `ca.pem` file is not available then the docke CLI client complains that ca.pem file si not available.
This is my experience with Docke CLI.